keyext.proofmanagement/src/main/java/org/key_project/proofmanagement/check/Node.java

```diff
@@ -0,0 +1,58 @@
+package org.key_project.proofmanagement.check;
+
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * Represents a vertex/node of a tree.
+ *
+ * @param <T> the type of the content contained by the node
+ * @author Wolfram Pfeifer
+ */
+public class Node<T extends Comparable<T>> implements Comparable<Node<T>> {
```

Comment:
Can this go into `key.util`?

Reply:
The class is not really general purpose: For a tree, in general the nodes are not needed to extend comparable. I tried to remove it, but for correct sorting in interplay with string template, it is needed (maybe it could be done differently, but I did not find an easy way). In addition, the compare method ignores the parent and child nodes at the moment. So I would suggest keeping the class here.